myhdl/_block.py

```diff
@@ -0,0 +1,250 @@
+        self._verifySubs()
+        self._updateNamespaces()
+        self.name = self.__name__ = mod.__name__ + '_' + str(mod.count)
+        self.verilog_code = self.vhdl_code = None
+        self.sim = None
+        if hasattr(mod, 'verilog_code'):
+            self.verilog_code = _UserVerilogCode(mod.verilog_code, self.symdict, mod.name,
+                                                 mod.modfunc, mod.sourcefile, mod.sourceline)
+        if hasattr(mod, 'vhdl_code'):
+            self.vhdl_code = _UserVhdlCode(mod.vhdl_code, self.symdict, mod.name,
+                                           mod.modfunc, mod.sourcefile, mod.sourceline)
+        self._conf_sim = {'trace': False}
+
+    def _verifySubs(self):
+        for inst in self.subs:
+            # print (inst.name, type(inst))
+            if not isinstance(inst, (_BlockInstance, _Instantiator, Cosimulation)):
+                raise BlockError(_error.ArgType)
+            if isinstance(inst, (_BlockInstance, _Instantiator)):
+                if not inst.modctxt:
+                    raise BlockError(_error.InstanceError % (self.mod.name, inst.callername))
+
+    def _updateNamespaces(self):
+        # dicts to keep track of objects used in Instantiator objects
+        usedsigdict = {}
+        usedlosdict = {}
+        for inst in self.subs:
+            # the symdict of a block instance is defined by
+            # the call context of its instantiations
+            if isinstance(inst, Cosimulation):
+                continue # ignore
+            if self.symdict is None:
+                self.symdict = inst.callinfo.symdict
+            if isinstance(inst, _Instantiator):
+                usedsigdict.update(inst.sigdict)
+                usedlosdict.update(inst.losdict)
+        if self.symdict is None:
+            self.symdict = {}
+        # Special case: due to attribute reference transformation, the
+        # sigdict and losdict from Instantiator objects may contain new
+        # references. Therefore, update the symdict with them.
+        # To be revisited.
+        self.symdict.update(usedsigdict)
+        self.symdict.update(usedlosdict)
+        # Infer sigdict and memdict, with compatibility patches from _extractHierarchy
+        for n, v in self.symdict.items():
+            if isinstance(v, _Signal):
+                self.sigdict[n] = v
+                if n in usedsigdict:
+                    v._markUsed()
+            if _isListOfSigs(v):
+                m = _makeMemInfo(v)
+                self.memdict[n] = m
+                if n in usedlosdict:
+                    m._used = True
+
+    def _inferInterface(self):
+        from myhdl.conversion._analyze import _analyzeTopFunc
+        intf = _analyzeTopFunc(self.mod.modfunc, *self.args, **self.kwargs)
+        self.argnames = intf.argnames
+        self.argdict = intf.argdict
+
+    # Public methods
+    # The puropse now is to define the API, optimizations later
+
+    def verifyConversion(self):
```

Comment:
whey do these need long names, wouldn't `verify` and `analyze` be sufficient?  And if not, should it follow the same naming pattern as `config_sim`?